src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs

```diff
@@ -332,6 +332,10 @@ public void ShowHelp(string commandName = null)
             else
             {
                 target = Commands.SingleOrDefault(cmd => string.Equals(cmd.Name, commandName, StringComparison.OrdinalIgnoreCase));
+                if (target == null)
+                {
+                    throw new ArgumentException(string.Format("TODO: Error: unexpected value '{0}' for option 'help'", commandName));
+                }
```

Comment:
consider passing the name of the parameter as the second parameter to the `ArgumentException` ctor. (Note, I am not feeling strongly about this - this is command line so probably parameter names don't matter)